### PR TITLE
Implement more robust isTTY check

### PR DESCRIPTION
**Pull request by kibertoad**
As per https://github.com/pinojs/pino-pretty/issues/220#issuecomment-912959528

### CODE REVIEW

**Comment by kibertoad** [comment]
@jorgebucaran What's up with GitHub actions? I see there is a CI flow defined, but it doesn't seem to be running.

**Comment by mcollina** [comment]
I can confirm this fixes the compatibility problems we have for pino transport

**Comment by jorgebucaran** [comment]
Wow, thanks! So, is this Node's canonical way to detect if you are in a pipe? Other than not being available inside worker_threads, are there any disadvantages to `process.stdout.isTTY`? Curious.

**Comment by jorgebucaran** [comment]
> @jorgebucaran What's up with GitHub actions? I see there is a CI flow defined, but it doesn't seem to be running.

It was configured to run against push only. It's fixed now. 🙆‍♂️

**Comment by kibertoad** [comment]
@jorgebucaran This discussion might be useful: https://github.com/nodejs/node/issues/28386

**Comment by mcollina** [comment]
@jorgebucaran is there an ETA for a release that would ship this?

File: index.js [hunk 1]
```diff
@@ -1,12 +1,13 @@
+const tty = require('tty');
```
Comment (jorgebucaran) [inline index.js:1]: Colorette is an ES module, so this should be changed to use `import`.

**Review by jorgebucaran — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (kibertoad) [reply index.js:1]: will that be compatible with current dual publishing of CJS and ES versions? do you autoreplace?

Comment (jorgebucaran) [reply index.js:1]: Nope, looks like the time to upgrade that script has finally come! I can take care of that after we merge this.

Comment (kibertoad) [reply index.js]: done

**Comment by jorgebucaran** [comment]
Thanks! I'll merge here, update the script and publish a new minor version shortly.

**Comment by jorgebucaran** [comment]
Released as 1.4.0.

**Comment by jorgebucaran** [comment]
@mcollina @kibertoad Is there any reason to check if `tty` is defined before using `tty.isatty()`? Should be defined, no?

```js
import * as tty from "tty"
 ...
const isCompatibleTerminal =
  tty && tty.isatty(1) && env.TERM && env.TERM !== "dumb"
```

**Comment by kibertoad** [comment]
@jorgebucaran It's a node-specific thing. There were prior bug reports from people using colorette outside of Node, I wanted to safeguard against that from the get go.

**Comment by jorgebucaran** [comment]
Gotcha, thanks. I still feel that this should be handled by the user's bundler / preprocessor (https://github.com/jorgebucaran/colorette/pull/64#issuecomment-922178724).

**Pull request merged**